wolfcrypt/src/aes.c

```diff
@@ -11527,8 +11672,9 @@ int wc_AesXtsEncrypt(XtsAes* xaes, byte* out, const byte* in, word32 sz,
             XMEMCPY(buf, in, sz);
 
             xorbuf(buf, tmp, AES_BLOCK_SIZE);
-            wc_AesEncryptDirect(aes, out - AES_BLOCK_SIZE, buf);
-            xorbuf(out - AES_BLOCK_SIZE, tmp, AES_BLOCK_SIZE);
+            ret = wc_AesEncryptDirect(aes, out - AES_BLOCK_SIZE, buf);
```

Comment:
See line above (around 11668):
if (sz >= AES_BLOCK_SIZE) {

RESTORE_VECTOR_REGISTERS() is not called before return.

Reply:
nice catch!  that bug is not new to this PR, but evidently, it's not detected by `DEBUG_VECTOR_REGISTER_ACCESS` testing.  would take some mighty fuzz testing to find that, if it's even reachable.

Reply:
(fix pushed)